homeassistant/components/xiaomi_aqara/__init__.py

```diff
@@ -61,20 +61,20 @@
 })
 
 
-GATEWAY_CONFIG_MAC_OPT = vol.Schema({
+GATEWAY_CONFIG = vol.Schema({
     vol.Optional(CONF_KEY):
         vol.All(cv.string, vol.Length(min=16, max=16)),
     vol.Optional(CONF_HOST): cv.string,
     vol.Optional(CONF_PORT, default=9898): cv.port,
     vol.Optional(CONF_DISABLE, default=False): cv.boolean,
 })
 
-GATEWAY_CONFIG_MAC_REQ = vol.Schema({
-    vol.Required(CONF_KEY):
-        vol.All(cv.string, vol.Length(min=16, max=16)),
-    vol.Optional(CONF_HOST): cv.string,
-    vol.Optional(CONF_PORT, default=9898): cv.port,
-    vol.Optional(CONF_DISABLE, default=False): cv.boolean,
+GATEWAY_CONFIG_MAC_OPTIONAL = GATEWAY_CONFIG.extend({
+    vol.Optional(CONF_MAC, default=None): vol.Any(GW_MAC, None)
```

Comment:
Since this is pop'ed with a default value, I don't think you need a default here(?)

Reply:
If I remove the default this configuration isn't valid anymore:

```
xiaomi_aqara:
  gateways:
    - key: ffffffffffffffff
      mac:
```

This configuration is still valid:

```
xiaomi_aqara:
  gateways:
    - key: ffffffffffffffff
```

Do you suggest a removal?

Reply:
To avoid a (small?) breaking change it shouldn't be changed cp. HA 0.91:

https://github.com/home-assistant/home-assistant/blob/0.90.2/homeassistant/components/xiaomi_aqara/__init__.py#L65

Reply:
Hmm, I thought that would be an explicit `None` and thus accepted. I may be wrong, I am not so strong with schemas.

I will tag @MartinHjelmare because he may like to comment. But this is such a minor issue, I think we should merge to get the fix into a beta and a follow-up PR can clean it up if we want.

Reply:
I don't think we should allow specifying an option key without a value. The only case where that is useful is where we want to activate components, ie the top level keys in the config. All other cases are just confusing and error prone.

Remove the default and mark it a breaking change.

Reply:
```py
vol.Optional(CONF_MAC): GW_MAC,
```

Reply:
And use `dict.get` when accessing that config key.